deps/rabbit/src/rabbit_amqp_session.erl

```diff
@@ -2628,6 +2627,11 @@ declare_queue(QNameBin,
             rabbit_core_metrics:queue_created(QName);
         {existing, _Q} ->
             ok;
+        {protocol_error, precondition_failed = _ErrorType, Reason, ReasonArgs} ->
```

Comment:
We could rename the returned error `{protocol_error, precondition_failed, Reason, ReasonArgs}` to something like `{error, queue_limit_exceeded, Reason, ReasonArgs}` so that each caller of `rabbit_queue_type:declare/2` can handle this error correctly?
"Blindly" mapping `precondition_failed` to `V_1_0_AMQP_ERROR_RESOURCE_LIMIT_EXCEEDED` is brittle. Returning a more correct error from `rabbit_queue_type:declare/2` makes more sense.